core-services/src/main/java/org/zalando/nakadi/cache/ChangesRegistry.java

```diff
@@ -0,0 +1,85 @@
+package org.zalando.nakadi.cache;
+
+import com.google.common.base.Charsets;
+import org.apache.curator.framework.api.CuratorWatcher;
+import org.apache.zookeeper.KeeperException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.exceptions.runtime.NakadiRuntimeException;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+import javax.annotation.Nullable;
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+@Component
+@Profile("!test")
+public class ChangesRegistry {
+    private final ZooKeeperHolder zk;
+    private static final String ET_CACHE_PATH = "/et_cache_changes";
+
+    @Autowired
+    public ChangesRegistry(final ZooKeeperHolder zk) {
+        this.zk = zk;
+    }
+
+    @PostConstruct
+    public void postConstruct() {
+        try {
+            zk.get().create().creatingParentsIfNeeded().forPath(ET_CACHE_PATH);
+        } catch (KeeperException.NodeExistsException ex) {
+            // Its all fine
+        } catch (Exception ex) {
+            throw new NakadiRuntimeException(ex);
```

Comment:
@rcillo Regarding this point - I think it makes sense not to retry, but just restart, as it may lead to  further problems during initialization.
I mean right now our code uses this kind of approach everywhere - we are retrying only in case if already started. During initialization - not. I thought to leave it like that to avoid confusion with other places

Reply:
Agreed.